### PR TITLE
[CINN]Enhance CacheKey hash logic by considering input dtypes

### PR DESCRIPTION
**Pull request by Aurelius84**
### PR types

Bug fixes
### PR changes

Others
### Describe

[CINN]Enhance CacheKey hash logic by considering input dtypes

### CODE REVIEW

File: paddle/fluid/framework/paddle2cinn/cinn_cache_key.cc [hunk 1]
```diff
@@ -85,10 +95,16 @@ size_t CinnCacheKey::Hash::operator()(const CinnCacheKey& key) const {
   for (const auto& name_shape : key.input_shapes_) {
     has_str << name_shape.first;
     has_str << std::hash<phi::DDim>()(name_shape.second);
+    PADDLE_ENFORCE_NE(key.input_dtypes_.find(name_shape.first),
```
Comment (thisjiang) [inline paddle/fluid/framework/paddle2cinn/cinn_cache_key.cc:100]: 第90行`CinnCacheKey::operator==`也应该加上`input_dtypes_ == other.input_dtypes_`

另外麻烦上面一行`has_str << std::hash<phi::DDim>()(name_shape.second);`直接改为`has_str << name_shape.second;`吧，更方便调试

Comment (thisjiang) [reply paddle/fluid/framework/paddle2cinn/cinn_cache_key.cc:100]: 我已经帮忙改了~

File: paddle/fluid/framework/paddle2cinn/transform_desc.cc [hunk 2]
```diff
@@ -97,6 +97,33 @@ ::cinn::frontend::paddle::cpp::VarDescAPI::Type TransformVarDataTypeToCinn(
 #undef SET_DATA_TYPE_CASE_ITEM
 }
 
+std::string VarDataTypeToString(
```
Comment (CtfGo) [inline paddle/fluid/framework/paddle2cinn/transform_desc.cc:100]: 补充单测

File: paddle/fluid/framework/paddle2cinn/transform_type.cc [hunk 3]
```diff
@@ -78,4 +79,47 @@ ::phi::DataType TransToPaddleDataType(const cinn_type_t& type) {
 #undef SET_TYPE_CASE_ITEM
 }
 
+std::string PaddleAttributeToString(const framework::Attribute& attr) {
```
Comment (CtfGo) [inline paddle/fluid/framework/paddle2cinn/transform_type.cc:82]: 补充单测

**Review by CtfGo — APPROVED** [review]
LGTM

**Pull request merged**